lib/puppet/util/windows/process.rb

```diff
@@ -97,6 +124,30 @@ def with_process_token(access, &block)
   end
   module_function :with_process_token
 
+  def get_process_image_name_by_pid(pid)
+    image_name = ""
+
+    open_process(PROCESS_QUERY_INFORMATION, FFI::WIN32_FALSE, pid) do |phandle|
+
+      FFI::MemoryPointer.new(:pdword, 1) do |exe_name_length_ptr|
```

Comment:
I think you actually want a pointer to a single dword, not a pointer to a single dword pointer - like this

```ruby
FFI::MemoryPointer.new(:dword, 1) do |exe_name_length_ptr|
```